src/hooks/useWalletAddress/useWalletAddress.ts

```diff
@@ -0,0 +1,26 @@
+import { KnownChainIds } from '@shapeshiftoss/types'
+import { useEffect, useState } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+export function useWalletAddress(defaultAddress: string = ''): string {
```

Comment:
this is specific to eth mainnet and shouldn't' be a hook, especially the way it's being consumed with the burn address

Reply:
Agreed. We don't need this. You'll need to:
- Check for ethSupport (grep for `supportsEthereumChain` for examples)
- Get the address either with chain-adapter, or with `accountSpecifiers` (grep for `selectFirstAccountSpecifierByChainId` - you might need to deserialize it with caip utils)

This can be done at component-level and doesn't need to be its own hook